src/main/scala/wdl4s/expression/FileEvaluator.scala

```diff
@@ -115,7 +115,7 @@ case class FileEvaluator(valueEvaluator: ValueEvaluator, coerceTo: WdlType = Wdl
       case a: Ast if a.isArrayOrMapLookup =>
         evalValue(a) match {
           case Success(f: WdlFile) => Success(Seq(f))
-          case _ => evaluateRecursive(a.getAttribute("rhs"))
+          case _ => evaluateRecursive(a.getAttribute("lhs"))
```

Comment:
The only think I can think of that not evaluating the `rhs` would do is fail to detect if someone uses files as keys for a map...
If we really want to catch this case we could still do something like
```scala
evaluateRecursive(a.getAttribute("lhs")) flatMap { lhs => evaluateRecursive(a.getAttribute("rhs")) map { rhs => lhs ++ rhs } }
```